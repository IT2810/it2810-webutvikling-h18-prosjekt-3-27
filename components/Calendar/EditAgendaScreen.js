import React, {PureComponent} from "react";
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  Button
} from "react-native";

class EditAgendaScreen extends PureComponent {
  static navigationOptions = {
    title: "Edit event"
  };

  state = {
    name: "",
    note: ""
  };

  willFocusSubscription = null;

  componentDidMount() {
    if (this.props.navigation) {
      this.willFocusSubscription = this.props.navigation.addListener(
        "willFocus",
        this.willFocus
        );
    }
  }

  componentWillUnmount() {
    if (this.willFocusSubscription) {
      this.willFocusSubscription.remove();
    }
  }

  willFocus = (payload) => {
    let item;
    if (this.props.navigation && (item = this.props.navigation.getParam("item"))) {
      this.setState({name: item.name, note: item.note});
    }
  };


  handleNameChange = (text) => {
    this.setState({name: text});
  };

  handleNoteChange = (text) => {
    this.setState({note: text});
  };

  handleSaveAgendaPress = () => {
    // button for adding a new agenda was clicked
    const callbackEditAgenda = this.props.navigation.getParam("editAgenda");
    if (!callbackEditAgenda)
      return;
    // create object for the new agenda
    const oldItem = this.props.navigation.getParam("item");
    const agenda = {
      ...oldItem,
      name: this.state.name,
      note: this.state.note
    };
    // pass it back to the calendar screen
    callbackEditAgenda(agenda);
    // then navigate back
    this.props.navigation.navigate("Calendar");
  };

  handleDeleteAgendaPress = () => {
    const callbackRemoveAgenda = this.props.navigation.getParam("removeAgenda");
    if (!callbackRemoveAgenda)
      return;
    // call delete callback
    const oldItem = this.props.navigation.getParam("item");
    callbackRemoveAgenda(oldItem);
    // then navigate back
    this.props.navigation.navigate("Calendar");
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.margins}>Name:</Text>
        <TextInput
          style={[styles.textInput, styles.margins]}
          value={this.state.name}
          underlineColorAndroid="transparent"
          placeholder={"Name"}
          onChangeText={this.handleNameChange}
          autoCorrect={false}
        />
        <Text style={styles.margins}>Description:</Text>
        <TextInput
          style={[styles.textInput, styles.margins]}
          value={this.state.note}
          multiline={true}
          numberOfLines={5}
          underlineColorAndroid="transparent"
          textAlignVertical="top"
          placeholder={"Enter description..."}
          onChangeText={this.handleNoteChange}
          autoCorrect={false}
        />
        <Button title={"Save event"} onPress={this.handleSaveAgendaPress}/>
        <View style={{margin: 10}}/>
        <Button title={"Delete event"} onPress={this.handleDeleteAgendaPress}/>
      </View>
    );
  }
}

export default EditAgendaScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "flex-start",
    backgroundColor: "white"
  },
  margins: {
    marginTop: 10,
    marginBottom: 10
  },
  textInput: {
    width: "80%",
    padding: 5,
    borderWidth: 1,
    borderColor: "#000"
  }
});
