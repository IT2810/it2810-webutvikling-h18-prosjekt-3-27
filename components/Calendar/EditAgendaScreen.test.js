import React from "react";
import renderer from "react-test-renderer";
import EditAgendaScreen from "./EditAgendaScreen";

it("renders correctly", () => {
  const tree = renderer
    .create(<EditAgendaScreen/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
