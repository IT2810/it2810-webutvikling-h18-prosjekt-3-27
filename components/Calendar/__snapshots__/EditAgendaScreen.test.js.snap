// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 5,
        "marginTop": 10,
      }
    }
  >
    Name:
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder="Name"
    style={
      Array [
        Object {
          "borderColor": "#606060",
          "borderRadius": 5,
          "borderWidth": 1,
          "padding": 5,
          "width": "80%",
        },
        Object {
          "marginBottom": 10,
          "marginTop": 5,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 5,
        "marginTop": 10,
      }
    }
  >
    Description:
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    multiline={true}
    numberOfLines={5}
    onChangeText={[Function]}
    placeholder="Enter description..."
    style={
      Array [
        Object {
          "borderColor": "#606060",
          "borderRadius": 5,
          "borderWidth": 1,
          "padding": 5,
          "width": "80%",
        },
        Object {
          "marginBottom": 10,
          "marginTop": 5,
        },
      ]
    }
    textAlignVertical="top"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "green",
            },
          ]
        }
      >
        Save event
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "margin": 10,
      }
    }
  />
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "green",
            },
          ]
        }
      >
        Delete event
      </Text>
    </View>
  </View>
</View>
`;
